files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java

```diff
@@ -739,6 +745,287 @@ public Mono<FileDetail> create(String clientCode, String uri, FilePart fp, Strin
+		if(targetPath[0]!=null) {
+			if (fp == null) {
+				if (ovr) {
+					targetPath[0] = Paths.get(this.getBaseLocation(),
+							this.resolvePathWithClientCode(filePath).getT1());
+				} else {
+					targetPath[0] = Paths.get(this.getBaseLocation(),
+							this.resolvePathWithClientCode(filePath).getT1());
+					
+					for (int i = 1; i <= 100; i++) {
+						int lastIndex = filePath.lastIndexOf(".");
+						String extension = filePath.substring(lastIndex);
+						Path newImagePath = targetPath[0].resolve(fileName + i + extension);
+						
+			            if (!Files.exists(newImagePath)) {
+			            	try {
+			                    Files.copy(targetPath[0], newImagePath, StandardCopyOption.REPLACE_EXISTING);
+			                    targetPath[0] = newImagePath;
+			                    break;
+			                    
+			                } catch (IOException e) {
+			                    Mono.error(e);
+			                }
+			            }
+			        }
+				}
+			}
+		}else {
+			Mono.empty();
+		}
+	}
+	
+	private Mono<FileDetail> updateAndProcessImage(FilePart fp, Path[] path, String fileName, boolean ovr,
+                                      ImageDetails imageDetails, String urlResourcePath, String clientCode){
+		return FlatMapUtil.flatMapMonoWithNull(
+
+				() -> {
+					if (fp != null) {
+						return fp.transferTo(path[0]);
+					} else {
+						return Mono.empty();
+					}
+				},
+
+				x -> {
+					return Mono.just(path[0].toFile());
+				},
+
+				(x, actualFile) -> {
+					try {
+						BufferedImage bufferedImage = ImageIO.read(actualFile);
+
+						BufferedImage resizedImage = resizeImage(bufferedImage, imageDetails.getWidth(),
+								imageDetails.getHeight());
+
+						BufferedImage flippedImage = resizedImage;
+						if (imageDetails.getFlipHorizontal().booleanValue()) {
+							BufferedImage flippedHorizontalImage = flipHorizontal(flippedImage);
+							flippedImage = null;
+							flippedImage = flippedHorizontalImage;
+						}
+						if (imageDetails.getFlipVertical().booleanValue()) {
+							BufferedImage flippedVerticalImage = flipVertical(flippedImage);
+							flippedImage = null;
+							flippedImage = flippedVerticalImage;
+						}
+
+						BufferedImage rotatedImage = rotateImage(flippedImage, imageDetails.getRotation(),
+								imageDetails.getBackgroundColor(), actualFile);
+
+						BufferedImage croppedImage = rotatedImage;
+						if (imageDetails.getCropAreaWidth() > 0 && imageDetails.getCropAreaHeight() > 0) {
+							croppedImage = cropImage(rotatedImage, imageDetails.getCropAreaX(),
+									imageDetails.getCropAreaY(), imageDetails.getCropAreaWidth(),
+									imageDetails.getCropAreaHeight());
+						}
+
+						return Mono.just(croppedImage);
+					} catch (IOException e) {
+						e.printStackTrace();
+						return Mono.empty();
+					}
+
+				},
+
+				(x, actualFile, resizedImage) -> {
+					try {
```

Comment:
Move the code inside try and catch block to a separate method to reduce the compexity of this method